### DISCUSSION TITLE
Regarding Setup of Sunbird Mobile App

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#5] [Sunbird ED configuration discussion] [answered]
Hello All,

I am trying to do a setup of SunbirdEd Mobile App , following this link
https://github.com/Sunbird-Ed/SunbirdEd-mobile-app , [read.me](http://read.me/) .

I am getting follwing errors while running the 'build.sh' file. 

Please check error images attached.

![Gradle5](https://user-images.githubusercontent.com/65657373/166231514-d051a5fb-1c74-4d67-9efc-ff7ec355d377.png)
![Gradle3](https://user-images.githubusercontent.com/65657373/166231560-9f26cded-9a84-4aed-ad62-ce51f0d17ffe.png)
![Gradle2](https://user-images.githubusercontent.com/65657373/166231621-c434e47e-5085-4a89-aca2-4aa3c724a6a4.png)
![Gradle1](https://user-images.githubusercontent.com/65657373/166231644-84311564-2d80-41f4-85a0-dce65bb90d44.png)

@coolbung @swayangjit  @vinukumar-vs @HarishGangula @Krishnaj20 @baljeetsingh @sharathprasad4tarento

**Comment by swayangjit** [comment, marked as answer]
@Xitija  Please comment out the[ prodiction section ](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/blob/f90c0e5fbc2922cb8c0d545532d980152c8069e6/buildConfig/build-extras.gradle#L145) in buildConfig/build-extras.gradle.

**Reply by Xitija** [reply]
I tried commenting it got an error , which I think is associated with production but I cannot figure out what it is.
![GradleMay2](https://user-images.githubusercontent.com/65657373/166240273-3ebe1108-0f7c-46bb-91f1-2a9635bf76d2.png)

**Reply by swayangjit** [reply]
@Xitija Try removing the node-moudles and then do a  `npm  i`. Then run `npm run ionic-build` if error persists then execute `npm i -D @angular-builders/custom-webpack`

**Reply by Xitija** [reply]
I tried this approaches 
1 : `npm i` and then `npm run ionic-build` got the following error.

![Gradmay31](https://user-images.githubusercontent.com/65657373/166410281-a28f151d-2547-4936-9092-82dbadd6d56a.png)
![Gradlmay32e](https://user-images.githubusercontent.com/65657373/166410299-0861474f-58e6-47f5-afbf-96da8b8d1480.png)
  

And 
2 : `npm i -D @angular-builders/custom-webpack` gives me this 
![GradleMay33](https://user-images.githubusercontent.com/65657373/166410450-ea14250c-5899-45d7-9c6e-8cc1d0300ff6.png)

but running `./build.sh ` still gives me an error 
![GradleMay34](https://user-images.githubusercontent.com/65657373/166410610-7008e276-15ec-4baa-93e3-cfe9025c1ef7.png)

@swayangjit

**Reply by Xitija** [reply]
Output of ionic build 
![iobu](https://user-images.githubusercontent.com/65657373/166449719-c6015770-12ea-4e3c-9fad-e7e1585bf0ed.png)

Output of build.sh
![buload](https://user-images.githubusercontent.com/65657373/166449790-c7b40a0e-1d4a-4fb8-8e48-b9a19d45504a.png)

**Reply by swayangjit** [reply]
@Xitija  Can you manually install loadash and check ?
Also, add a snapshot of ionic info and node -v ?

**Reply by Xitija** [reply]
Snapshot of ionic info and node -v 
![ver1](https://user-images.githubusercontent.com/65657373/166631568-873c9bba-bdee-4f90-bdaf-450976fd3ac5.png)
![ver2](https://user-images.githubusercontent.com/65657373/166631582-622846f8-de77-48dc-8c92-ae5e9e1108e7.png)

Used these commands for installing loadlash
`npm i -g npm`
`npm i --save lodash`
Its output is
![ver3](https://user-images.githubusercontent.com/65657373/166632067-6de5dcc2-572f-4808-886c-5a8fed6d19b9.png)
![ver4](https://user-images.githubusercontent.com/65657373/166632082-30cd22dd-a46d-49f4-85fe-73072713a292.png)

` npm run ionic-build` still gives me an error
![errrr1](https://user-images.githubusercontent.com/65657373/166632841-bd0f1094-b38e-4bc2-92b0-84587b32e782.png)
![errrr2](https://user-images.githubusercontent.com/65657373/166632859-24b6837a-030e-4fbd-b18b-3256d25bd5b7.png)

`npm i ` is giving me this 
![npmerror](https://user-images.githubusercontent.com/65657373/166632906-b09444a7-4b82-40a1-95e8-5a95a1b190c3.png)

@swayangjit

**Reply by swayangjit** [reply]
@Xitija There is some issue in build-extras.gradle where you have missed a "}"  and also mention your node version , I think by mistake you have posted your ionic version

**Reply by Xitija** [reply]
node version is 12.21.0
@itsvick
Helped me create the build

**Reply by swayangjit** [reply]
@Xitija is it working now ?

**Reply by Xitija** [reply]
Yes it is working . We made changes to the build-extras.gradle , google-services.json
cd /home/ttpl-rt-032/Android/Sdk/build-tools/33.0.0-rc3 && mv d8 dx && cd lib && mv d8.jar dx.jar used this command
and lastly npm run ionic-build

cc : @baljeetsingh

**Reply by pramodkvarma** [reply]
@swayangjit do we need these fixes back in the scripts? How do we make sure docs reflect these for others?

**Reply by swayangjit** [reply]
@pramodkvarma  The error that we are getting here has nothing to do with Sunbird scripts. This is more to do with how android SDK is installed in the system.  We are planning to add some FAQ section in README for all these kinds of common issues where we will provide the solution or else provide a StackOverflow link.

**Reply by pramodkvarma** [reply]
Ok thanks for clarifying. Yes do update FAQ and point to this discussion URL.

**Reply by swayangjit** [reply]
@pramodkvarma Ok sure.
